lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
```

Comment:
I'm not concerned with save an in-memory hash lookup when resetting a password. I just want it to be as clear as possible. 

with regards to `blank?` or `empty?`, `blank?` seems safer. If for whatever reason that's `nil`, `blank?` still works as expected while `empty?` does not.